### PR TITLE
[WIP] LiDAR Update

### LINKED ISSUE #17: Lidar Device Issue
Have you tried running this same project on a Lidar-enabled device?

I tried it on my iPad Pro 2020 and it did not perform well with locating planes.
It would fail to change orientation to match the new surface.

I also tried it with a prior version of the project, commit 88112e6, and it has the same issues.

I think this could have something to do with the raycasting, but `sceneUnderstanding.options` does not include `.collisions`, so it should not be affecting raycast results.

I did also use
```
self.debugOptions.insert(.showAnchorGeometry)
```
To make sure that planes were being detected, and they were.

### PR DESCRIPTION
**Pull request by maxxfrazer**
[WORK IN PROGRESS]

All responses from the ARRaycastQuery were estimated geometries when using lidar.
But after switching to only allow existing plane geometries by default it seems to work as well as you'd expect a lidar device to!

@Reality-Dev try it out and let me know if it works better for you, there's another line I may delete too, found in the comments

#17

### CODE REVIEW

File: Sources/FocusEntity/FocusEntity.swift [hunk 1]
```diff
@@ -261,6 +265,7 @@ open class FocusEntity: Entity, HasAnchoring, HasFocusEntity {
     if self.currentAlignment != .none {
       // It is ready to move over to a new surface.
       recentFocusEntityPositions.append(position)
+      performAlignmentAnimation(to: raycastResult.worldTransform.orientation)
```
Comment (maxxfrazer) [inline Sources/FocusEntity/FocusEntity.swift:268]: Need to test this on non-LiDAR device, might make things all messed up, but on LiDAR with estimated planes allowed it looks pretty cool! Might make it a FocusEntityComponent option

**Comment by maxxfrazer** [comment]
This video shows it working great across two vertical planes and on the ceiling plane, still need to try it out on non-lidar later but this seems to have done the trick

https://user-images.githubusercontent.com/5754073/106921130-9b1f1380-6703-11eb-90fc-2437a5937199.mov

**Comment by Reality-Dev** [comment]
The video looks great. Nice work!

File: Sources/FocusEntity/FocusEntity.swift (hunk 1 above)
Comment (maxxfrazer) [reply Sources/FocusEntity/FocusEntity.swift:268]: looks good to me on non-LiDAR, I'll leave it in

**Comment by maxxfrazer** [comment]
All looks good with and without LiDAR, thanks for spotting those issues! Going to merge now

**Pull request merged**